plsync/mlabconfig.py

```diff
@@ -45,11 +65,185 @@ def parse_flags():
+
+
+def export_experiment_records_v4(output, sites, experiment, decoration=''):
+  comment(output, '%s v4%s' % (experiment.dnsname(), (
+      ' decorated' if decoration else '')))
+  for site in sites:
+    # TODO: change site['nodes'] to a pre-sorted list type.
+    for node in sorted(site['nodes'].values(), key=lambda n: n.hostname()):
+        # TODO: should sitenames exclude mlab4's?
+        write_a_record(output, experiment.sitename(node, decoration),
+                       experiment.ipv4(node))
+        write_a_record(output, experiment.recordname(node, decoration),
+                       experiment.ipv4(node))
+
+
+def export_experiment_records_v6(output, sites, experiment, decoration=''):
+  comment(output, '%s v6%s' % (experiment.dnsname(), (
+      ' decorated' if decoration else '')))
+  for site in sites:
+    # TODO: change site['nodes'] to a pre-sorted list type.
+    for node in sorted(site['nodes'].values(), key=lambda n: n.hostname()):
+      # TODO: should sitenames exclude mlab4's?
+      if (node.ipv6_is_enabled() and experiment.ipv6(node)):
+        write_aaaa_record(output, experiment.sitename(node, decoration),
+                          experiment.ipv6(node))
+        write_aaaa_record(output, experiment.recordname(node, decoration),
+                          experiment.ipv6(node))
+
+
+def export_mlab_zone_records(output, sites, experiments):
+    export_router_and_switch_records(output, sites)
+    export_pcu_records(output, sites)
+    export_server_records(output, sites)
+    export_experiment_records(output, sites, experiments)
+
+
+def get_revision(revision_path):
+    """Returns a two digit revision number as a string.
+
+    Everytime the same revision_path is provided, the previous revision number
+    is incremented by one and returned. However, the revision number does not
+    increase beyond "99".
+
+    Args:
+        revision_path: str, the full path to a temporary file to read previous
+            and store latest revision number.
+
+    Returns:
+        str, a two digit revision number, e.g. "00", "01", etc, up to "99".
+    """
+    n = 0
+    if os.path.exists(revision_path):
+        with open(revision_path) as f:
+            n = int(f.read())
+            if n < 99:
+                n += 1
+            else:
+                logging.error('Revision is too large to increase!')
+
+    revision = '%02d' % n
+    with open(revision_path, 'w') as f:
+        f.write(revision)
+
+    return revision
+
+
+def serial_rfc1912(ts):
+    """Returns an rfc1912 style serial id (YYYYMMDDnn) for a DNS zone file."""
+    # RFC1912 (http://www.ietf.org/rfc/rfc1912.txt) recommends 'nn' as the
+    # revision. However, identifying and incrementing this value is a manual,
+    # error prone step. Instead, we save a temporary daily sequence counter.
+    serial_prefix = time.strftime('%Y%m%d', ts)
+    return serial_prefix + get_revision(os.path.join('/tmp', serial_prefix))
```

Comment:
This works, but also means that this script can _only_ ever be run on a single machine. This may be fine, but I wanted to point this out (not that it wasn't already obvious).